### PR TITLE
Setting File Access Lists

### PR DESCRIPTION
**Pull request by ucheNkadiCode**
Responded to user request for more information on how to add file permissions to access files on a host machine from within a container. I went ahead and revamped the document.

**Important**: I chose to change the title, so I have also since renamed the root file. This means that the URL will change. I need to know AS SOON AS the link is active because this will affect our dockerfile scaffolding https://github.com/microsoft/vscode-docker/issues/2747.

Question: Should the DateApproved metadata be updated to indicate when the doc was last edited? Or should we keep it as April 2020?
Thank you!

### CODE REVIEW

File: docs/containers/debug-python.md
Comment (bwateratmsft) [inline docs/containers/debug-python.md:235]: Is `.md.md` right?

**Comment by gregvanl** [comment]
@ucheNkadiCode We will need to add a redirect on the website for this file rename so that people who already have bookmarked the old URL will go to the new URL. I can add the redirect in the private website repo before this PR is merged.
Per the DateApproved, yes, it is good to update the date so users know the topic is getting attention and is not stale.

**Comment by gregvanl** [comment]
@ucheNkadiCode @ghogen I'll do the merge once everyone is signed off so I can control the timing relative to the website redirect.

**Review by ucheNkadiCode — COMMENTED** [review]
> @ucheNkadiCode We will need to add a redirect on the website for this file rename so that people who already have bookmarked the old URL will go to the new URL. I can add the redirect in the private website repo before this PR is merged.
> Per the DateApproved, yes, it is good to update the date so users know the topic is getting attention and is not stale.

Thank you @gregvanl! I have updated the DateApproved. I appreciate the help with the website redirect! We are gearing up to have our aka.ms links point to the new link whenever that goes live. Please either you or @ghogen send me the exact link so I don't mess it up!

File: docs/containers/quickstart-python.md
Comment (ghogen) [inline docs/containers/quickstart-python.md:58]:
```suggestion
1. If **Python: Django** or **Python: Flask** was selected, specify app port for local development. Django defaults to port 8000, while Flask defaults to port 5000; however, any unused port will work. We recommend selecting port 1024 or above to mitigate security concerns from [running as a root user](/docs/containers/python-configure-containers.md).
```

File: docs/containers/python-configure-containers.md
Comment (ghogen) [inline docs/containers/python-configure-containers.md:150]:
```suggestion
In the previous example, we showed you how to add permissions to a file or folder on the container as a non-root user. However, if you are trying to access a folder **on the host machine** from within the container as a non-root user, the user ID or group ID in the container must have access to the files on the host. To solve this issue in Linux, you might need to set file access control lists (setfacl).
```

File: docs/containers/python-configure-containers.md
Comment (ghogen) [inline docs/containers/python-configure-containers.md:146]:
```suggestion
> **Note**: This is just one example of how to add permissions in a container. There are many ways to do so, and it is your responsibility give the least permission possible to specific files and folders.
```

**Review by ghogen — APPROVED** [review]
Approving with a few small suggestions.

**Comment by gregvanl** [comment]
@ucheNkadiCode The update has been published. 
The new URL is https://code.visualstudio.com/docs/containers/python-configure-containers
I've added a redirect from /docs/containers/python-user-rights to /docs/containers/python-configure-containers so that no bookmarks will be broken.

File: docs/containers/python-configure-containers.md
Comment (TheYarin) [inline docs/containers/python-configure-containers.md:162]: why powershell and not shell/bash?

Comment (TheYarin) [reply docs/containers/python-configure-containers.md:162]: same for line 173

File: docs/containers/python-configure-containers.md
Comment (TheYarin) [inline docs/containers/python-configure-containers.md:160]: Maybe it's worth mentioning how to enter a container's commandline

Comment (TheYarin) [inline docs/containers/python-configure-containers.md:160]: Are you sure the user/group ID will be the same if the container is removed and re-created? That's a common scenario when operating with `docker-compose up` &  `docker-compose down`

File: docs/containers/python-configure-containers.md
Comment (bwateratmsft) [reply docs/containers/python-configure-containers.md:162]: That's just telling the docs site how to syntax highlight this chunk of code. Those commands are actually shell/bash commands.

Comment (TheYarin) [reply docs/containers/python-configure-containers.md:162]: Yeah I know, this might seem a little petty but why not mark the block specifically as `bash`/`sh`? I'm pretty sure whichever markdown engine is used, it supports this language.

File: docs/containers/python-configure-containers.md
Comment (ucheNkadiCode) [reply docs/containers/python-configure-containers.md:160]: @bwateratmsft or @TheYarin Would it be sufficient to say 

1. From the **Containers** pane in the Docker View, right click on your running container, select **Attach Shell**, and run one of these commands:

Comment (TheYarin) [reply docs/containers/python-configure-containers.md:160]: hmm, I think the user will be doing all this permissions configuration in a production environment, so docker actions from within the developer's VSCode might not be relevant.

(Unless managing your remote docker host from within VSCode is really easy, of which I might not be familiar with)

Comment (ucheNkadiCode) [reply docs/containers/python-configure-containers.md:160]: @karolz-ms or @bwateratmsft, do you know about this?

Comment (bwateratmsft) [reply docs/containers/python-configure-containers.md:160]: I don't know for sure, but I would think that the uid/gid would stay the same, since it comes from the image, not the container. I think?

File: docs/containers/python-configure-containers.md
Comment (karolz-ms) [reply docs/containers/python-configure-containers.md:162]: ... or  `shell`  I agree with @TheYarin

File: docs/containers/python-configure-containers.md
Comment (karolz-ms) [reply docs/containers/python-configure-containers.md:160]: Good catch @TheYarin Best practice is to use and explicit UID in this case: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user

Comment (TheYarin) [reply docs/containers/python-configure-containers.md:160]: Yeah, I'm expecting  this scenario will mostly occur as a remote container. This _might_ happen in the development environment if it's a linux environment (when mounting a volume on my Windows dev environment it didn't cause any permissions errors), but honestly, I got a feeling that a lot of developers don't properly test their application in a containerized setup on their dev machine, because the developer probably has all the dependencies set up already so it's easier for him to just run the app outside the container.

But considering what @karolz-ms said below about the best practice regarding UID, I think mentioning how to enter a container's commandline is kind of irrelevant.

**Pull request merged**
